lib/pos_blockchain.go

```diff
@@ -262,15 +262,44 @@ func (bc *Blockchain) validateBlockHeight(desoBlock *MsgDeSoBlock) error {
 // validateBlockView validates the view for a given block. First, it checks that
 // the view is greater than the latest committed block view. If not,
 // we return an error indicating that we'll never accept this block. Next,
-// it checks that the view is less than or equal to the latest uncommitted block.
+// it checks that the view is less than or equal to its parent.
 // If not, we return an error indicating that we'll want to add this block as an
 // orphan. Then it will check if that the view is exactly one greater than the
 // latest uncommitted block if we have an regular vote QC. If this block has a
 // timeout QC, it will check that the view is at least greater than the latest
 // uncommitted block's view + 1.
 func (bc *Blockchain) validateBlockView(desoBlock *MsgDeSoBlock) error {
-	// TODO: Implement me
-	return errors.New("IMPLEMENT ME")
+	committedBlock, _ := bc.getHighestCommittedBlock()
+	if committedBlock == nil {
+		// This is an edge case we'll never hit in practice since all the PoW blocks
+		// are committed.
+		return errors.New("validateBlockView: No committed blocks found")
+	}
+	// Validate that the view is greater than the latest committed block view.
+	if desoBlock.Header.ProposedInView <= committedBlock.Header.ProposedInView {
+		return RuleErrorPoSBlockViewEarlierThanCommittedBlock
+	}
+	// Validate that the view is greater than the latest uncommitted block.
+	parentBlock, exists := bc.blockIndex[*desoBlock.Header.PrevBlockHash]
+	if !exists {
+		// Note: this should never happen as we only call this function after
+		// we've validated that all ancestors exist in the block index.
+		return RuleErrorMissingParentBlock
+	}
+	// If our current block has a vote QC, then we need to validate that the
+	// view is exactly one greater than the latest uncommitted block.
+	if desoBlock.Header.ValidatorsTimeoutAggregateQC.isEmpty() {
+		if desoBlock.Header.ProposedInView != parentBlock.Header.ProposedInView+1 {
+			return RuleErrorPoSVoteBlockViewNotOneGreaterThanParent
+		}
+	} else {
+		// If our current block has a timeout QC, then we need to validate that the
+		// view is strictly greater than the latest uncommitted block's view.
+		if desoBlock.Header.ProposedInView <= parentBlock.Header.ProposedInView {
+			return RuleErrorPoSTimeoutBlockViewNotGreaterThanParent
+		}
+	}
```

Comment:
These are great sanity checks for the blocks view. But they're not the full set of checks we need to perform on the view later down in the ProcessBlockPoS flow. Later on, you'll need to compare the block's view against the node's current view, and to perform the safeVote predicate (https://github.com/deso-protocol/hotstuff_pseudocode/blob/main/fast_hotstuff_bls.go#L430).

IMO, it'll be cleaner to move this code into the `validateBlockIntegrity` function so it's clear these are just sanity checks.

Reply:
Will introduce the rest of those validations in a separate PR. All these validation functions are called from `validateDeSoBlockPoS` and wanted to keep each validation function a reasonable size. We can refactor later if we want and rejigger the tests, but I'd prefer to keep them broken up for now.